src/components/activities/MainPage.client.tsx

```diff
@@ -19,59 +22,91 @@ const getPageSize = (width: number) => {
 const MainPageClient = ({ initialData }: { initialData: Activities }) => {
   const innerWidth = useWindowWidth();
   const [pageSize, setPageSize] = useState(ITEM_DEFAULT_PAGESIZE);
+  const [searchQuery, setSearchQuery] = useState('');
   const loadMoreRef = useRef<HTMLDivElement>(null);
 
+  console.log('이니셜 데이터:', initialData)
   useEffect(() => {
     if (innerWidth) setPageSize(getPageSize(innerWidth));
-  }, [innerWidth]);
+  }, [innerWidth,]);
 
   const { data, fetchNextPage, hasNextPage, isFetchingNextPage, isError } =
     useSuspenseInfiniteQuery<
       Activities,
       Error,
       InfiniteData<Activities, unknown>,
-      [string, number],
+      [string, string, number],
       number | null
     >({
-      queryKey: ['activities', pageSize],
-      queryFn: ({ pageParam = null }) =>
-        fetchServerData<Activities>({
+      queryKey: ['activities', searchQuery, pageSize],
+      queryFn: ({ pageParam = null }) => {
+        const queryParams: { [key: string]: string | number } = {
+          method: 'cursor',
+          size: pageSize,
+        };
+        if (pageParam !== null) { queryParams.cursorId = pageParam; }
+        if (searchQuery) { queryParams.keyword = searchQuery; }
+
+        return fetchServerData<Activities>({
           path: '/activities',
-          query: { method: 'cursor', cursorId: pageParam ?? undefined, size: pageSize },
-        }),
-      initialPageParam: null, // 첫 요청 시 cursorId = null
+          query: queryParams,
+        });
+      },
+      initialPageParam: null,
       getNextPageParam: lastPage => {
-        // 응답에 cursorId가 있으면 다음 요청에 사용
         return lastPage.activities.length > 0 ? lastPage.cursorId : undefined;
       },
-      initialData: { pages: [initialData], pageParams: [null] }, // 서버에서 받은 데이터로 초기 캐시 세팅
     });
 
-  // 무한 스크롤 트리거
   useIntersectionObserver({
     target: loadMoreRef,
     onIntersect: fetchNextPage,
     enabled: hasNextPage,
   });
 
   const isFetchingMore = hasNextPage && isFetchingNextPage;
+  const allActivities = data?.pages.flatMap(page => page.activities) ?? [];
+
+  const handleSearch = (query: string) => {
+    // 검색어 상태를 업데이트하여 TanStack Query가 새로운 검색 결과를 자동으로 불러오게 합니다.
+    setSearchQuery(query);
+  };
 
   return (
     <>
-      <ActivityList data={data} />
-      <div ref={loadMoreRef} className='min-h-10'>
-        {isError && (
-          <p className='pb-16 text-center'>
-            목록 불러오기에 실패했습니다.
-            <button className='ml-2 underline underline-offset-4' onClick={() => fetchNextPage()}>
-              다시 시도
-            </button>
-          </p>
+      <SearchBarClient onSearch={handleSearch} />
```

Comment:
메인 페이지에서 서버에서 가져오고 있는 체험 initialdata 를 프롭으로 이 컴포넌트에 전달해서 이 컴포넌트 안에서는 그걸 useQuery 로 캐싱하는 전략을 사용하면 어떤가요?
```suggestion
      <SearchBarClient onSearch={handleSearch} data={initialData} onShearchResult={handleSearchResult}/>
      <>
```